job-server/src/main/scala/spark/jobserver/JobManagerActor.scala

```diff
@@ -121,6 +124,12 @@ class JobManagerActor(contextConfig: Config, daoActor: ActorRef) extends Instrum
     Option(jobContext).foreach(_.stop())
   }
 
+  // Handle external kill events (e.g. killed via YARN)
+  override def onApplicationEnd(event: SparkListenerApplicationEnd) {
+    logger.info("Got Spark Application end event, stopping job manger.")
```

Comment:
I asked a question related to this in the mailing list but have not got any response.

http://apache-spark-user-list.1001560.n3.nabble.com/Application-kill-from-UI-do-not-propagate-exception-td28539.html